processor/stash/stash.go

```diff
@@ -234,9 +237,9 @@ func (st *HandleT) readErrJobsLoop(ctx context.Context) {
 			}
 			//NOTE: sending custom val filters array of size 1 to take advantage of cache in jobsdb.
 			toQuery := errDBReadBatchSize
-			retryList := st.errorDB.GetToRetry(jobsdb.GetQueryParamsT{CustomValFilters: []string{""}, JobCount: toQuery, IgnoreCustomValFiltersInQuery: true})
+			retryList := st.errorDB.GetToRetry(jobsdb.GetQueryParamsT{CustomValFilters: []string{""}, IgnoreCustomValFiltersInQuery: true, JobsLimit: toQuery, PayloadSizeLimit: payloadLimit})
 			toQuery -= len(retryList)
-			unprocessedList := st.errorDB.GetUnprocessed(jobsdb.GetQueryParamsT{CustomValFilters: []string{""}, JobCount: toQuery, IgnoreCustomValFiltersInQuery: true})
+			unprocessedList := st.errorDB.GetUnprocessed(jobsdb.GetQueryParamsT{CustomValFilters: []string{""}, IgnoreCustomValFiltersInQuery: true, JobsLimit: toQuery, PayloadSizeLimit: payloadLimit})
```

Comment:
Don't we need to check if we have hit the limits before fetching unprocessed?

Reply:
Yes, to be efficient though it needs further refactoring so that instead of a slice of jobs we return some more information, e.g. 

```
type JobsResult struct {
  Jobs          []*JobT
  EventsCount   int
  PayloadCount  int64
  LimitsReached bool
}
```
For the time being this should be enough though, but will open another task to implement it in the next sprint.